File: app/src/main/java/org/kindone/willingtodo/persistence/TaskPersistenceProvider.java
package org.kindone.willingtodo.persistence;

import org.kindone.willingtodo.data.Task;

import java.util.List;

/**
 * Created by kindone on 2016. 12. 13..
 */
public interface TaskPersistenceProvider {
    int getVersion();

    List<Task> getTasksOfContextOrderedByPriority(long contextId);

    List<Task> getTasksOfContextOrderedByWillingness(long contextId);

    Task createTask(Task task);

    void updateTask(Task task);

    void swapPriorityOfTasks(long id1, long id2);

    void swapWillingnessOfTasks(long id1, long id2);

    void deleteTask(long id);
}

File: app/src/main/java/org/kindone/willingtodo/persistence/TaskPersistenceProvider.kt
package org.kindone.willingtodo.persistence

import org.kindone.willingtodo.data.Task

/**
 * Created by kindone on 2016. 12. 13..
 */
interface TaskPersistenceProvider {
    val version: Int

    fun getTasksOfContextOrderedByPriority(contextId: Long): List<Task>

    fun getTasksOfContextOrderedByWillingness(contextId: Long): List<Task>

    fun createTask(task: Task): Task

    fun updateTask(task: Task)

    fun swapPriorityOfTasks(id1: Long, id2: Long)

    fun swapWillingnessOfTasks(id1: Long, id2: Long)

    fun deleteTask(id: Long)
}
